tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+
+        assert generated_mat.dtype == "complex128"
+        assert qnp.allclose(generated_mat, expected_mat)
+
+
+class TestIntegration:
+    def test_measurement_process_expval(self):
+        """Test SProd class instance in expval measurement process."""
+        dev = qml.device("default.qubit", wires=2)
+        sprod_op = SProd(1.23, qml.Hadamard(1))
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.PauliX(0)
+            return qml.expval(sprod_op)
+
+        exp_val = my_circ()
+        true_exp_val = qnp.array(1.23 / qnp.sqrt(2))
+        assert qnp.allclose(exp_val, true_exp_val)
+
+    def test_measurement_process_var(self):
+        """Test SProd class instance in var measurement process."""
+        dev = qml.device("default.qubit", wires=2)
+        sprod_op = SProd(1.23, qml.Hadamard(1))
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.PauliX(0)
+            return qml.var(sprod_op)
+
+        var = my_circ()
+        true_var = qnp.array(1.23**2 / 2)
+        assert qnp.allclose(var, true_var)
+
+    def test_measurement_process_probs(self):
+        """Test SProd class instance in probs measurement process raises error."""  # currently can't support due to bug
+        dev = qml.device("default.qubit", wires=2)
+        sprod_op = SProd(1.23, qml.Hadamard(1))
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.PauliX(0)
+            return qml.probs(op=sprod_op)
+
+        with pytest.raises(
+            QuantumFunctionError,
+            match="Symbolic Operations are not supported for " "rotating probabilities yet.",
+        ):
+            my_circ()
+
+    def test_measurement_process_sample(self):
+        """Test SProd class instance in sample measurement process raises error."""  # currently can't support due to bug
+        dev = qml.device("default.qubit", wires=2)
+        sprod_op = SProd(1.23, qml.Hadamard(1))
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.PauliX(0)
+            return qml.sample(op=sprod_op)
+
+        with pytest.raises(
+            QuantumFunctionError, match="Symbolic Operations are not supported for sampling yet."
+        ):
+            my_circ()
+
+    def test_differentiable_measurement_process(self):
+        """Test that the gradient can be computed with a SProd op in the measurement process."""
+        sprod_op = SProd(100, qml.Hadamard(0))
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, grad_method="best")
+        def circuit(weights):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.RX(weights[2], wires=1)
+            return qml.expval(sprod_op)
+
+        weights = qnp.array([0.1, 0.2, 0.3], requires_grad=True)
+        grad = qml.grad(circuit)(weights)
+
+        true_grad = 100 * qnp.array([-7.05928859e-02, 1.38777878e-17, 2.77555756e-17])
+        assert qnp.allclose(grad, true_grad)
+
```

Comment:
Migth also be worth testing the differentiability of the scalar.  It works with backdrop:
```
@qml.qnode(qml.device('default.qubit', wires=1))
def circuit(x):
    return qml.expval(qml.ops.s_prod(x, qml.PauliZ(0)))
```

Reply:
Added!